### DISCUSSION TITLE
Error when opening zipfiles requests. zipfile.BadZipFile: File is not a zip file

### DISCUSSION THREAD

**Discussion by Tyarel8** [encode/httpx#2213] [Potential Issue] [answered]
After attempting to donwload and extract a zipfile using httpx
```
rz = httpx.get(url)
with zipfile.ZipFile(BytesIO(rz.content)) as z:
         z.extractall(DOWNLOAD_PATH)
```
it throws the following error `zipfile.BadZipFile: File is not a zip file`

However, using the exact same code but with the requests module works as intended
```
rz = requests.get(url)
with zipfile.ZipFile(BytesIO(rz.content)) as z:
         z.extractall(DOWNLOAD_PATH)
```

**Comment by lovelydinosaur** [comment]
Could you provide an example that we're able to reproduce?

Specifically, including the URL you're using would help here.

**Reply by Tyarel8** [reply]
```
import zipfile
from io import BytesIO
import httpx

r = httpx.get("https://golang.org/dl/go1.17.3.windows-amd64.zip")
with zipfile.ZipFile(BytesIO(r.content)) as zf:
    zf.extractall()
```

**Reply by nidemidovich** [reply]
Well, when I tried this GET request, I got 301 response code.

**Comment by Tyarel8** [comment, marked as answer]
Solved, I didn't notice that httpx didn't follow redirects by default, that was why it was working with the request module but not with httpx. (follow_redirects=True)

**Comment by snjjay** [comment]
GZIP, renaming, tracing, and 7zip expressions are the most basic ways to overcome the problem (Zipfile.badzipfile: file is not a zip) in Python. In fact, these two options necessitate the installation of new modules. [Read this](https://www.kodlogs.net/326/zipfile-badzipfile-file-is-not-a-zip-file)
